Sources/Swift/Integrations/SessionReplay/SentryVideoFrameProcessor.swift

```diff
@@ -46,46 +46,49 @@ class SentryVideoFrameProcessor {
     }
 
     func processFrames(videoWriterInput: AVAssetWriterInput, onCompletion: @escaping (Result<SentryRenderVideoResult, Error>) -> Void) {
-        SentrySDKLog.debug("[Session Replay] Video writer input is ready, status: \(videoWriter.status)")
-        guard videoWriter.status == .writing else {
-            SentrySDKLog.error("[Session Replay] Video writer is not writing anymore, cancelling the writing session, reason: \(videoWriter.error?.localizedDescription ?? "Unknown error")")
-            videoWriter.inputs.forEach { $0.markAsFinished() }
-            videoWriter.cancelWriting()
-            return onCompletion(.failure(videoWriter.error ?? SentryOnDemandReplayError.errorRenderingVideo))
-        }
-        guard frameIndex < videoFrames.count else {
-            SentrySDKLog.debug("[Session Replay] No more frames available to process, finishing the video")
-            return finishVideo(frameIndex: self.frameIndex, onCompletion: onCompletion)
-        }
-
-        let frame = videoFrames[frameIndex]
-        defer {
-            // Increment the frame index even if the image could not be appended to the pixel buffer.
-            // This is important to avoid an infinite loop.
-            frameIndex += 1
-        }
-        guard let image = UIImage(contentsOfFile: frame.imagePath) else {
-            return
-        }
-
-        SentrySDKLog.debug("[Session Replay] Image at index \(frameIndex) is ready, size: \(image.size)")
-        guard lastImageSize == image.size else {
-            SentrySDKLog.debug("[Session Replay] Image size has changed, finishing video")
-            return finishVideo(frameIndex: self.frameIndex, onCompletion: onCompletion)
-        }
-        lastImageSize = image.size
-
-        let presentTime = SentryOnDemandReplay.calculatePresentationTime(
-            forFrameAtIndex: frameIndex,
-            frameRate: frameRate
-        ).timeValue
-        guard currentPixelBuffer.append(image: image, presentationTime: presentTime) else {
-            SentrySDKLog.error("[Session Replay] Failed to append image to pixel buffer, cancelling the writing session, reason: \(String(describing: videoWriter.error))")
-            videoWriter.inputs.forEach { $0.markAsFinished() }
-            videoWriter.cancelWriting()
-            return onCompletion(.failure(videoWriter.error ?? SentryOnDemandReplayError.errorRenderingVideo))
+        while videoWriterInput.isReadyForMoreMediaData {
```

Comment:
`m: At first glance, this looks like something that could easily lead to an endless loop. It would be great to add a comment that calling `isReadyForMoreMediaData` in while is the recommended approach of https://developer.apple.com/documentation/avfoundation/avassetwriterinput/requestmediadatawhenready(on:using:)

After looking at the tests and the code docs, it seems like not doing this while loop will lead to missing frames. If this fixes a bug, it would be great to document this here as well.